coral/src/app/features/topics/request/TopicPromotionRequest.tsx

```diff
@@ -0,0 +1,282 @@
+              <NativeSelect
+                name="environment"
+                labelText={"Environment"}
+                required
+                readOnly
+              >
+                <Option
+                  key={targetEnvironment.name}
+                  value={targetEnvironment.name}
+                >
+                  {targetEnvironment.name}
+                </Option>
+              </NativeSelect>
+            ) : (
+              <NativeSelect.Skeleton />
+            )}
+          </Box>
+          <Box>
+            <Box paddingY={"l1"}>
+              <Divider />
+            </Box>
+            <TextInput<Schema>
+              name={"topicname"}
+              labelText="Topic name"
+              placeholder={generateTopicNameDescription(
+                targetEnvironment?.params
+              )}
+              required={true}
+              readOnly
+            />
+            <Box.Flex gap={"l1"}>
+              <Box grow={1} width={"1/2"}>
+                {targetEnvironment !== undefined ? (
+                  <SelectOrNumberInput
+                    name={"topicpartitions"}
+                    label={"Topic partitions"}
+                    max={targetEnvironment.params?.maxPartitions}
+                    required={true}
+                  />
+                ) : (
+                  <Input.Skeleton />
+                )}
+              </Box>
+              <Box grow={1} width={"1/2"}>
+                {targetEnvironment !== undefined ? (
+                  <SelectOrNumberInput
+                    name={"replicationfactor"}
+                    label={"Replication factor"}
+                    max={targetEnvironment.params?.maxRepFactor}
+                    required={true}
+                  />
+                ) : (
+                  <Input.Skeleton />
+                )}
+              </Box>
+            </Box.Flex>
+          </Box>
+          <Box>
+            <Box paddingY={"l1"}>
+              <Divider />
+            </Box>
+            <AdvancedConfiguration name={"advancedConfiguration"} />
+          </Box>
+
+          <Box>
+            <Box paddingY={"l1"}>
+              <Divider />
+            </Box>
+            <Box.Flex gap={"l1"}>
+              <Box grow={1} width={"1/2"}>
+                <Textarea<Schema>
+                  name="description"
+                  labelText="Description"
+                  rows={5}
+                  required={true}
+                  readOnly
+                />
+              </Box>
+              <Box grow={1} width={"1/2"}>
+                <Textarea<Schema>
+                  name="remarks"
+                  labelText="Message for approval"
+                  rows={5}
+                />
+              </Box>
+            </Box.Flex>
+          </Box>
+          {promoteIsError && (
+            <Box marginBottom={"l1"} role="alert">
+              <Alert type="error">{parseErrorMsg(promoteError)}</Alert>
```

Comment:
> In all other forms, we show the error at the top of the form, not at the end, this should be consistent.

I actually think it's a mistake to not show it where I put it here. Because often forms are long enough that you scroll down, and the alert is then not visible! It is the case in particular for this form.

This is what it looks like with the box at the top:


https://github.com/aiven/klaw/assets/20607294/9e71d1f4-f762-4b6e-98b3-bd5b4345c9fa

we may want two follow up tickets the, for dividers and for the placement of the error box.

Reply:
>  and the alert is then not visible! It is the case in particular for this form.

Yep, that's a problem in all/most of our forms, because we don't know which screen width the user has and where they are on the page. Usually, it is recommended (in regards to accessibility) to scroll to the content of an error message when it appears on a page. That's not a pattern that is introduced in console right now, but there are discussions about it. Form design, validation and error handling is definitely a topic where usability (for all users) can be improved still :D

The pattern with placing <Alert /> right above <Form> is the same as console follows, so we should also adhere to that for now.

Reply:
Ok 😞 

Will change it back.

Reply:
[05956b1](https://github.com/aiven/klaw/pull/1420/commits/05956b1cce5c815b0a5d5d0e85f90a8b69655982)